docs/topics/wasm/wasm-debugging.md

```diff
@@ -1,148 +1,114 @@
 
-## Run the application
+   `WasmDemo/composeApp/src/wasmJsMain/kotlin/wasm.project.demo.wasmdemo`
 
-1. In IntelliJ IDEA, open the **Gradle** tool window by selecting **View** | **Tool Windows** | **Gradle**.
+2. Click on the line numbers to set breakpoints on the code that you want to inspect.
 
-   > You need at least Java 11 as your Gradle JVM for the tasks to load successfully.
-   >
-   {style="note"}
+   ![Set breakpoints](wasm-breakpoints-intellij.png){width=650}
 
-2. In **wasmdemo** | **Tasks** | **kotlin browser**, select and run the **wasmJsBrowserDevelopmentRun** task.
+3. In the list of run configurations, click **Edit configurations** | **+** | **Gradle** and
+   select `wasmJsBrowserDevelopmentRun` from the **Run** list.
+4. Run the code in debug mode by clicking the debug icon at the top of the screen. Make sure
+   you run the `wasmJsBrowserDevelopmentRun` configuration.
 
-   ![Run the Gradle task](wasm-gradle-task-window.png){width=450}
+   ![Run in debug mode](wasm-debug-run-configurations.png){width=600}
 
-   Alternatively, you can run the following command in the terminal from the `WasmDemo` root directory:
+   Once the application starts, it opens in a new browser window, and the **Debug** panel appears automatically in IntelliJ IDEA.
 
-   ```bash
-   ./gradlew wasmJsBrowserDevelopmentRun
-   ```
+   ![Compose app in browser](wasm-composeapp-browser.png){width=600}
 
-3. Once the application starts, open the following URL in your browser:
+### Inspect your application
 
-   ```bash
-   http://localhost:8080/
-   ```
+> If you're [debugging in your browser](#debug-in-your-browser), you can follow the same steps for inspecting your application.
+>
+{style="note"}
 
-   > The port number can vary because the 8080 port may be unavailable. You can find the actual port number printed
-   > in the Gradle build console.
-   >
-   {style="tip"}
+1. In the application's browser window, click on the **Click me!** button to interact with the application. This action triggers the execution of the
+   code, and the debugger pauses when the execution reaches a breakpoint.
 
-   You see a "Click me!" button. Click it:
+2. In the debugging pane, use the debugging control buttons to inspect variables and code execution at the breakpoints:
+    * ![Step over](wasm-debug-step-over.png){width=30}{type="joined"} Step over to execute the current line and pause on the next line.
+    * ![Step into](wasm-debug-step-into.png){width=30}{type="joined"} Step into to investigate a function more deeply.
+    * ![Step out](wasm-debug-step-out.png){width=30}{type="joined"} Step out to execute the code until it exits the current function.
 
-   ![Click me](wasm-composeapp-browser-clickme.png){width=550}
+3. Check the **Threads & Variables** pane to trace the sequence of function calls and pinpoint the location of any errors.
 
-   Now you see the Compose Multiplatform logo:
+   ![Check Threads & Variables](wasm-debug-panes-intellij.png){width=700}
```

Comment:
Please update screenshort with the latest IDEA Nightly or EAPCandidate

Reply:
I tested it again in 2025.3, and I get the same. Any other different screenshot you can provide?

<img width="1562" height="1013" alt="Screenshot 2025-09-03 at 7 54 24 PM" src="https://github.com/user-attachments/assets/05a45d81-b5ba-4b8f-8ef0-dfc4d0c8d5ad" />
<img width="1255" height="541" alt="Screenshot 2025-09-03 at 7 54 34 PM" src="https://github.com/user-attachments/assets/ac0cca4d-5d18-4fb6-ae3a-3ff6322d1063" />